fv3core/stencils/d2a2c_vect.py

```diff
@@ -382,72 +379,110 @@ class DGrid2AGrid2CGridVectors:
     Fortran name d2a2c_vect
     """
 
-    def __init__(self, grid, namelist, dord4):
-        if namelist.grid_type >= 3:
-            raise Exception("unimplemented grid_type >= 3")
-        self.grid = grid
+    def __init__(
+        self,
+        grid_indexing: GridIndexing,
+        cosa_s,
+        cosa_u,
+        cosa_v,
+        rsin_u,
+        rsin_v,
+        rsin2,
+        dxa,
+        dya,
+        sin_sg1,
+        sin_sg2,
+        sin_sg3,
+        sin_sg4,
+        nested: bool,
+        grid_type: int,
+        # TODO: remove npx and layout, and in doing so
+        # remove MPI-grid dependence of outputs
+        npx: int,
+        layout: Tuple[int, int],
+        dord4: bool,
+    ):
+        self._cosa_s = cosa_s
+        self._cosa_u = cosa_u
+        self._cosa_v = cosa_v
+        self._rsin_u = rsin_u
+        self._rsin_v = rsin_v
+        self._rsin2 = rsin2
+        self._dxa = dxa
+        self._dya = dya
+        self._sin_sg1 = sin_sg1
+        self._sin_sg2 = sin_sg2
+        self._sin_sg3 = sin_sg3
+        self._sin_sg4 = sin_sg4
+
+        if grid_type >= 3:
+            raise NotImplementedError("unimplemented grid_type >= 3")
         self._big_number = 1e30  # 1e8 if 32 bit
-        nx = self.grid.ie + 1  # grid.npx + 2
-        ny = self.grid.je + 1  # grid.npy + 2
-        i1 = self.grid.is_ - 1
-        j1 = self.grid.js - 1
+        nx = grid_indexing.iec + 1  # grid.npx + 2
+        ny = grid_indexing.jec + 1  # grid.npy + 2
+        i1 = grid_indexing.isc - 1
+        j1 = grid_indexing.jsc - 1
         id_ = 1 if dord4 else 0
         pad = 2 + 2 * id_
-        npt = 4 if not self.grid.nested else 0
-        if npt > self.grid.nic - 1 or npt > self.grid.njc - 1:
+        npt = 4 if not nested else 0
+        if npt > grid_indexing.domain[0] - 1 or npt > grid_indexing.domain[1] - 1:
             npt = 0
         self._utmp = utils.make_storage_from_shape(
-            self.grid.domain_shape_full(add=(1, 1, 1)),
-            self.grid.full_origin(),
+            grid_indexing.max_shape,
+            grid_indexing.origin_full(),
         )
         self._vtmp = utils.make_storage_from_shape(
-            self.grid.domain_shape_full(add=(1, 1, 1)), self.grid.full_origin()
+            grid_indexing.max_shape, grid_indexing.origin_full()
         )
 
-        js1 = npt + OFFSET if self.grid.south_edge else self.grid.js - 1
-        je1 = ny - npt if self.grid.north_edge else self.grid.je + 1
-        is1 = npt + OFFSET if self.grid.west_edge else self.grid.isd
-        ie1 = nx - npt if self.grid.east_edge else self.grid.ied
+        js1 = npt + OFFSET if grid_indexing.south_edge else grid_indexing.jsc - 1
+        je1 = ny - npt if grid_indexing.north_edge else grid_indexing.jec + 1
+        is1 = npt + OFFSET if grid_indexing.west_edge else grid_indexing.isd
+        ie1 = nx - npt if grid_indexing.east_edge else grid_indexing.ied
```

Comment:
so fun. luckily all of this is removed in the single stencil version of this code